internal/server/iigointernal/orchestration.go

```diff
@@ -76,64 +91,82 @@ func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.C
 	executiveBranch.loadClientPresident(presidentPointer)
 	legislativeBranch.loadClientSpeaker(speakerPointer)
 
-	// Withdraw the salaries
-	presidentWithdrawSuccess := judicialBranch.withdrawPresidentSalary(g)
-	judgeWithdrawSuccess := legislativeBranch.withdrawJudgeSalary(g)
-	speakerWithdrawSuccess := executiveBranch.withdrawSpeakerSalary(g)
+	// Pay salaries into budgets
 
-	// Handle the lack of resources
-	if !presidentWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since President has no resources to spend"
-	}
-	if !judgeWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since Judge has no resources to spend"
-	}
-	if !speakerWithdrawSuccess {
-		returnWithdrawnSalariesToCommonPool(g, &executiveBranch, &legislativeBranch, &judicialBranch)
-		return false, "Could not run IIGO since Speaker has no resources to spend"
+	errorJudicial := judicialBranch.sendPresidentSalary()
+	errorLegislative := legislativeBranch.sendJudgeSalary()
+	errorExecutive := executiveBranch.sendSpeakerSalary()
+	// Throw error
+	if errorJudicial != nil || errorLegislative != nil || errorExecutive != nil {
+		return false, "Cannot pay IIGO salary"
```

Comment:
For the time being this is the intended behavior, yes

Reply:
Hi, could we just get a confirm on this, I remember that for MVP we specifically put in rollback functions to prevent this behaviour. @Spruudis @ezgimez is this intended?